api/users_test.go

```diff
@@ -242,17 +279,28 @@ func TestServer_UpdateUserHandler(t *testing.T) {
 			name:     "handles user not found error",
 			sendBody: `{"id": "` + id.String() + `", "email": "test@example.com", "firstname": "Bob", "lastname": "Smith"}`,
 
-			wantUserDto:   &dto.UserDto{ID: id.String(), Email: "test@example.com", Firstname: "Bob", Lastname: "Smith"},
+			wantUpdateUser: &store.UpdateUser{
+				ID:         id,
+				CreateUser: store.CreateUser{Email: "test@example.com", Firstname: "Bob", Lastname: "Smith"},
+			},
 			wantUserModel: nil,
 			returnErr:     store.ErrNotFound,
 
 			expectedStatusCode: http.StatusNotFound,
 		},
 		{
-			name:     "handles internal server error",
-			sendBody: `{"id": "` + id.String() + `", "email": "test@example.com", "firstname": "Bob", "lastname": "Smith"}`,
-
-			wantUserDto:   &dto.UserDto{ID: id.String(), Email: "test@example.com", Firstname: "Bob", Lastname: "Smith"},
+			name: "handles internal server error",
+			sendBody: `{
+				"id": "` + id.String() + `",
+				"email": "test@example.com",
+				"firstname": "Bob",
+				"lastname": "Smith"
+			}`,
```

Comment:
I assume this valid body is reused a few times, you can store it in a variable at the top of the test func that you call `validBody`. Makes it easier to read the test cases imo